gapic/src/main/com/google/gapid/widgets/LoadingIndicator.java

```diff
@@ -148,12 +156,20 @@ public Widget createWidgetWithRefresh(Composite parent) {
 
   /**
    * Widget that shows the loading indicator while loading and is blank once done.
+   * Can optionally show an image when done.
    */
   public class Widget extends Canvas implements Loadable, Repaintable {
-    private boolean loading = false;
+    private Image successImage;
+    private Image failureImage;
+
+    protected boolean loading = false;
+    protected boolean status = false;
 
     public Widget(Composite parent, boolean showRefresh) {
       super(parent, SWT.DOUBLE_BUFFERED);
+      successImage = null;
```

Comment:
Get rid of this constructor and simply call your new one with either `null, null`, or `refresh, refresh` above.